packages/app/src/cli/services/dev/processes/dev-session/dev-session.ts

```diff
@@ -61,12 +62,34 @@ export class DevSession {
 
   /**
    * Handle the app event, after validating the event it might trigger a dev session update.
+   * If an update is already in progress, it will queue the event to be processed after
+   * the current update completes.
    * @param event - The app event
    */
   private async onEvent(event: AppEvent) {
     const eventIsValid = await this.validateAppEvent(event)
     if (!eventIsValid) return
-    this.currentBundleController?.abort()
+
+    this.appEventsQueue.push(event)
+
+    if (!this.updateInProgress) await this.processEvents()
+  }
+
+  /**
+   * Process an app event by bundling extensions and uploading them.
+   * After completion, it will check if there are any pending events in the queue,
+   * consolidate them into a single event, and process it.
+   * @param event - The app event to process
+   */
+  private async processEvents() {
+    this.updateInProgress = true
+    const events = [...this.appEventsQueue]
+
+    this.appEventsQueue = []
+
+    const event = this.consolidateAppEvents(events)
+    if (!event) return
+
     this.statusManager.setMessage('CHANGE_DETECTED')
     await this.updatePreviewURL(event)
     await this.logger.logExtensionEvents(event)
```

Comment:
Should this be happening in the serialised/consolidated block? Maybe information is lost in this case

Reply:
this is already the consolidated event, it has information about all extensions that changed since the last update so `await this.logger.logExtensionEvents(event)` will properly print all extensions that are being updated.